src/Nethermind/Nethermind.Core/TransactionReceipt.cs

```diff
@@ -58,8 +58,8 @@ public class TxReceipt
         /// </summary>
         public byte StatusCode { get; set; }
         public long BlockNumber { get; set; }
-        public KeccakStructRef BlockHash;
-        public KeccakStructRef TxHash;
+        public ref readonly ValueKeccak BlockHash;
```

Comment:
Interesting, does this mean it only store pointer to the value? Is this only valid for ref struct?

Reply:
Only valid for `ref struct` can do `ref readonly` for a `readonly struct` and perhaps more confucingly can do `readonly ref readonly` for one that can only be set in the constructor (for example in a `readonly ref struct`)

Reply:
Yes, is pointer (ref) to value